jnigen/test/summary_test.dart

```diff
@@ -0,0 +1,324 @@
+    expect(specific.typeParams, hasLength(0));
+    expect(specific.superclass, isNotNull);
+    final sAnc = specific.superclass!.type as DeclaredType;
+    expect(sAnc.params[0].type, isA<DeclaredType>());
+    expect(sAnc.params[0].type.name, equals('java.lang.String'));
+
+    final generic = classes.getClass('inheritance', 'GenericDerivedClass');
+    expect(generic.typeParams, hasLength(1));
+    expect(generic.typeParams[0].name, equals('T'));
+    expect(generic.typeParams[0].bounds.map((b) => b.name).toList(),
+        ['java.lang.CharSequence']);
+    expect(generic.superclass, isNotNull);
+    final gAnc = generic.superclass!.type as DeclaredType;
+    expect(gAnc.params[0].type, isA<TypeVar>());
+    expect(gAnc.params[0].type.name, equals('T'));
+  });
+
+  test('constructor is included', () {
+    final example = classes.getExampleClass();
+    void assertOneCtorExistsWithArity(int arity, List<String> paramTypes) {
+      final arityCtors = example.methods
+          .where((m) => m.name == '<init>' && m.params.length == arity)
+          .toList();
+      expect(arityCtors, hasLength(1));
+      final ctor = arityCtors[0];
+      expect(ctor.params.map((p) => p.type.name), equals(paramTypes));
+    }
+
+    assertOneCtorExistsWithArity(0, []);
+    assertOneCtorExistsWithArity(1, ['int']);
+    assertOneCtorExistsWithArity(2, ['int', 'boolean']);
+    assertOneCtorExistsWithArity(3, ['int', 'boolean', 'java.lang.String']);
+  });
+
+  test('Overloaded methods', () {
+    final methods = classes
+        .getExampleClass()
+        .methods
+        .where((m) => m.name == 'overloaded')
+        .toList();
+    expect(methods, hasLength(5));
+    final signatures =
+        methods.map((m) => m.params.map((p) => p.type.name).toList()).toList();
+    expect(
+        signatures,
+        containsAll(const [
+          <String>[],
+          ['int'],
+          ['int', 'java.lang.String'],
+          ['java.util.List'],
+          ['java.util.List', 'java.lang.String'],
+        ]));
+  });
+
+  test('Declaration type (class vs interface vs enum)', () {
+    final example = classes.getExampleClass();
+    expect(example.declKind, DeclKind.classKind);
+    final myInterface = classes.getClass('interfaces', 'MyInterface');
+    expect(myInterface.declKind, DeclKind.interfaceKind);
+    final color = classes.getClass('simple_package', 'Color');
+    expect(color.declKind, DeclKind.enumKind);
+  });
+
+  test('Enum values', () {
+    final example = classes.getExampleClass();
+    expect(example.values, anyOf(isNull, isEmpty));
+    final color = classes.getClass('simple_package', 'Color');
+    const expectedEnumValues = {
+      'RED',
+      'BLUE',
+      'BLACK',
+      'GREEN',
+      'YELLOW',
+      'LIME'
+    };
+    expect(color.values?.toSet(), expectedEnumValues);
+  });
+
+  test('Static final field values', () {
+    final example = classes.getExampleClass();
+    expect(example.getField("ON").defaultValue, equals(1));
```

Comment:
nit: You can also write `expect(x, equals(y))` like `expect(x, y)`.

Reply:
Does it matter? Should I change this?

Reply:
It doesn't matter. Just shorter I guess.